src/proto/extension.rs

```diff
@@ -32,7 +99,46 @@ impl Decode for SessionBind {
     }
 }
 
-#[derive(Debug, Clone)]
+impl Encode for SessionBind {
```

Comment:
Ha. I've pulled the latest changes in my agent and made a duplcate in https://github.com/wiktor-k/ssh-agent-lib/pull/48

Reply:
or you just cherry picked the commit ><

Reply:
Great timing! I cherry picked your changes in https://github.com/wiktor-k/ssh-agent-lib/commit/d761362f51987caa5b23ff2b3f6bf393ddf65019 as my first attempt at the destination constraints left a lot to be desired 😅